src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -360,6 +360,10 @@ internal void FunctionEnvironmentReloadResponse(FunctionEnvironmentReloadRespons
             _workerChannelLogger.LogDebug("Received FunctionEnvironmentReloadResponse from WorkerProcess with Pid: '{0}'", _rpcWorkerProcess.Id);
 
             LogWorkerMetadata(res.WorkerMetadata);
+
+            _workerConfig.Description.DefaultRuntimeVersion = _workerConfig.Description.DefaultRuntimeVersion ?? res.WorkerMetadata.RuntimeVersion;
+            _workerConfig.Description.DefaultRuntimeName = _workerConfig.Description.DefaultRuntimeName ?? res.WorkerMetadata.RuntimeName;
```

Comment:
Question: So we are only updating these properties if they are null. Env reload response will have the correct runtime value (after specialization). In that case, shouldn't we just override irrespective of existing value being null or not?

Reply:
We don't know for sure if all workers are sending the version and runtime name as part of metadata response. If we override this info and worker is only providing this info as part of workerConfig then we would be overriding it with null.